src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Entities/CacheableEntity.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+
+namespace OrchardCore.Entities;
+
+public class CacheableEntity : ICacheableEntity
+{
+    private JObject _properties = new();
+
+    private Dictionary<string, object> _cache { get; } = new Dictionary<string, object>();
+
+    public JObject Properties
+    {
+        get
+        {
+            return _properties;
+        }
+        set
+        {
+            _properties = value ?? new JObject();
+            _cache.Clear();
+        }
+    }
+
+    public void Forget(string key)
+    {
+        AssertNotNull(key);
```

Comment:
```suggestion
        ArgumentNullException.ThrowIfNullOrEmpty(key)
```

Reply:
Yes, better. done.

Reply:
@hishamco `ArgumentNullException.ThrowIfNullOrEmpty` is part of .net7 and won't work with .net6. Maybe we can change this when .net8 is released and we stop support for .net6